sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -22,103 +25,194 @@ namespace Azure.Messaging.EventHubs
+            }
+
+            remove
+            {
+                if (_processEventAsync != value)
+                {
+                    throw new ArgumentException("TODO: value does not exist.");
+                }
+
+                EnsureNotRunningAndInvoke(() => _processEventAsync = default);
+            }
         }
 
         /// <summary>
-        ///   Responsible for processing unhandled exceptions thrown while this processor is running.
+        ///   The event responsible for processing unhandled exceptions thrown while this processor is running.
         ///   Implementation is mandatory.
         /// </summary>
         ///
-        public Func<ProcessorErrorContext, ValueTask> ProcessErrorAsyncHandler
+        public event Func<ProcessErrorEventArgs, Task> ProcessErrorAsync
         {
-            get => _processErrorAsyncHandler;
-            set => EnsureNotRunningAndInvoke(() => _processErrorAsyncHandler = value);
+            add
+            {
+                if (_processErrorAsync != default)
+                {
+                    throw new NotSupportedException("TODO: no more than one event.");
+                }
+
+                EnsureNotRunningAndInvoke(() => _processErrorAsync = value);
+            }
+
+            remove
+            {
+                if (_processErrorAsync != value)
+                {
+                    throw new ArgumentException("TODO: value does not exist.");
+                }
+
+                EnsureNotRunningAndInvoke(() => _processErrorAsync = default);
+            }
         }
 
         /// <summary>
         ///   The fully qualified Event Hubs namespace that the processor is associated with.  This is likely
         ///   to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
         /// </summary>
         ///
-        public override string FullyQualifiedNamespace { get; }
+        public string FullyQualifiedNamespace { get; }
 
         /// <summary>
         ///   The name of the Event Hub that the processor is connected to, specific to the
         ///   Event Hubs namespace that contains it.
         /// </summary>
         ///
-        public override string EventHubName { get; }
+        public string EventHubName { get; }
 
         /// <summary>
         ///   The name of the consumer group this event processor is associated with.  Events will be
         ///   read only in the context of this group.
         /// </summary>
         ///
-        public override string ConsumerGroup { get; }
+        public string ConsumerGroup { get; }
 
         /// <summary>
-        ///   A unique name used to identify this event processor.
+        ///   Indicates whether or not this event processor is currently running.
         /// </summary>
         ///
-        public override string Identifier { get; }
+        public bool IsRunning
+        {
+            get
+            {
+                if (_isRunning.HasValue)
```

Comment:
I think you didn't get what `_isRunning` is used for (I agree that's a bit confusing). Originally, I intended to have:
```
public bool IsRunning => ActiveLoadBalancingTask != null && !ActiveLoadBalancingTask.Completed;
```

However, we want to make IsRunning have a `protected set` so it can be mocked. For this reason, I have included the private `_isRunning` member (take a look at the description):

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L54-L55

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L192-L209

If it's `null`, it means it hasn't been mocked yet and we'll return the actual running status. Once it's set via the `protected set` method, it can never be set to `null` again, and the mocked value is used instead.

I think that's useful in a mocking scenario. We wouldn't want the processor overwriting the mocked value set by the user.

Reply:
Renamed `_isRunning` to `_isRunningMock`. Let me know if you think that's a bad idea.

Reply:
Fair enough.   Maybe `_isRunningOverride` with a `<remark>` that just calls out this is intended to override default behavior for mocking?    Definitely can defer until after this PR.